Documentation/ApiOverview/RequestHandling/Index.rst

```diff
@@ -300,6 +300,149 @@ identifier. This will circumvent the risk of circularity::
+PSR-17_ response factory interface (the concrete TYPO3 implementation is injected as constructor
+dependency) to create a new PSR-7_ response object:
+
+.. code-block:: php
+
+    use Psr\Http\Message\ResponseFactoryInterface;
+    use Psr\Http\Message\ResponseInterface;
+    use Psr\Http\Message\ServerRequestInterface;
+    use Psr\Http\Server\MiddlewareInterface;
+    use Psr\Http\Server\RequestHandlerInterface;
+
+    class StatusCheckMiddleware implements MiddlewareInterface
+    {
+        /** @var ResponseFactoryInterface */
+        private $responseFactory;
+
+        public function __construct(ResponseFactoryInterface $responseFactory)
+        {
+            $this->responseFactory = $responseFactory;
+        }
+
+        public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+        {
+            if ($request->getRequestTarget() === '/check') {
+                $data = ['status' => 'ok'];
+                $response = $this->responseFactory->createResponse()
+                    ->withHeader('Content-Type', 'application/json; charset=utf-8');
+                $response->getBody()->write(json_encode($data));
+                return $response;
+            }
+            return $handler->handle($request);
+        }
+    }
+
+.. _request-handling-psr-18:
+
+Executing HTTP Requests in Middlewares
+======================================
+
+The PSR-18_ HTTP Client is intended to be used by PSR-15_ request handlers in order to perform HTTP
+requests based on PSR-7_ message objects without relying on a specific HTTP client implementation.
+
+PSR-18_ consists of a client interface and three exception interfaces:
+
+- :php:`\Psr\Http\Client\ClientInterface`
+- :php:`\Psr\Http\Client\ClientExceptionInterface`
+- :php:`\Psr\Http\Client\NetworkExceptionInterface`
+- :php:`\Psr\Http\Client\RequestExceptionInterface`
+
+Request handlers shall use dependency injection to retrieve the concrete implementation
+of the PSR-18_ HTTP client interface :php:`\Psr\Http\Client\ClientInterface`.
+
+The PSR-18_ HTTP Client interface is provided by `psr/http-client` and may be used as
+dependency for services in order to perform HTTP requests using PSR-7_ request objects.
+PSR-7_ request objects can be created with the :ref:`PSR-17 Request Factory interface<request-handling-psr-17>`.
+
+.. note::
+
+   This does not replace the currently available Guzzle wrapper
+   :php:`\TYPO3\CMS\Core\Http\RequestFactory->request()`, but is available as a framework
+   agnostic, more generic alternative. The PSR-18 interface does not allow to pass request
+   specific guzzle options. But global options defined in :php:`$GLOBALS['TYPO3_CONF_VARS']['HTTP']`
+   are taken into account as GuzzleHTTP is used as backend for this PSR-18 implementation.
+   The concrete implementations is internal and will be replaced by a native guzzle PSR-18
+   implementation once it is available.
```

Comment:
```suggestion
   This does not replace the currently available Guzzle wrapper
   :php:`\TYPO3\CMS\Core\Http\RequestFactory->request()`, but is available as a more generic, 
   framework-agnostic alternative. The PSR-18 interface does not allow you to pass 
   request-specific guzzle options. But global options defined in :php:`$GLOBALS['TYPO3_CONF_VARS']['HTTP']`
   are taken into account because GuzzleHTTP is used as the backend for this PSR-18 implementation.
   The concrete implementation is internal and will be replaced by a native guzzle PSR-18
   implementation once it is available.
```